lib/awilixManager.ts

```diff
@@ -1,11 +1,19 @@
-import type { AwilixContainer } from 'awilix'
+import {
+  type AwilixContainer,
+  type BuildResolver,
+  type BuildResolverOptions,
+  type Constructor,
+  type DisposableResolver,
+  asClass,
+} from 'awilix'
+import type { Resolver } from 'awilix/lib/resolvers'
 
 declare module 'awilix' {
   // eslint-disable-next-line @typescript-eslint/no-unused-vars
   interface ResolverOptions<T> {
     asyncInit?: boolean | string
     asyncInitPriority?: number // lower means it gets initted earlier
-    asyncDispose?: boolean | string | ((instance: T) => Promise<unknown>)
+    asyncDispose?: boolean | string | (<U extends T>(instance: U) => Promise<unknown>)
```

Comment:
Without this change it wasn't possible to pass a class that would extend the original class as a resolver override.

E. g. something like this:
```
consumerErrorResolver: asClass(FakeConsumerErrorResolver, SINGLETON_CONFIG),
```

Reply:
Nice! I was just writing up a repro for this! https://codesandbox.io/p/devbox/awilix-ts-interface-j5sll2

Reply:
wahahaha, awesome!